doc/rbd/rbd-mirroring.rst

```diff
@@ -0,0 +1,233 @@
+option to the ``rbd`` command. 
+
+Alternatively, the journaling feature can be dynamically enabled on
+pre-existing RBD images.  To enable journaling with ``rbd``, specify
+the ``feature enable`` command, the pool and image name, and the feature name::
+
+        rbd feature enable {pool-name}/{image-name} {feature-name}
+
+For example::
+
+        rbd --cluster local feature enable image-pool/image-1 journaling
+
+.. note:: The journaling feature is dependent on the exclusive-lock feature.  If
+   the exclusive-lock feature is not already enabled, it should be enabled prior
+   to enabling the journaling feature.
+
+.. tip:: You can enable journaling on all new images by default by adding
+   ``rbd default features = 125`` to your Ceph configuration file.
+
+Enable Image Mirroring
+----------------------
+
+If the mirroring is configured in ``image`` mode for the image's pool, then it
+is necessary to explicitly enable mirroring for each image within the pool.
+To enable mirroring for a specific image with ``rbd``, specify the
+``mirror image enable`` command along with the pool and image name::
+
+        rbd mirror image enable {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image enable image-pool/image-1
+
+Disable Image Mirroring
+-----------------------
+
+To disable mirroring for a specific image with ``rbd``, specify the
+``mirror image disable`` command along with the pool and image name::
+
+        rbd mirror image disable {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image disable image-pool/image-1
+
+Image Promotion and Demotion
+----------------------------
+
+An image can only be primary within a single Ceph cluster.  Attempting to
+access a non-primary image will result in an error.  In a failover scenario,
+access to the primary image should be stopped (e.g. power down the VM or remove
+the associated drive from a VM), demote the current primary image, promote
+the new primary image, and resume access to the image on the alternate
+cluster.
+
+To demote an image to secondary with ``rbd``, specify the ``mirror image demote``
+command along with the pool and image name::
+
+        rbd mirror image demote {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster local mirror image demote image-pool/image-1
+
+To promote an image to primary with ``rbd``, specify the ``mirror image promote``
+command along with the pool and image name::
+
+        rbd mirror image promote {pool-name}/{image-name}
+
+For example::
+
+        rbd --cluster remote mirror image promote image-pool/image-1
+
+.. tip:: Since the primary / secondary status is per-image, it is possible to
+   have two clusters split the IO load and stage failover / failback.
+.. note:: Promotion can be forced using the ``--force`` option.  This will result
+   in a split-brain scenario between the two peers and the image will no longer
+   be in-sync until a `force resync command`_ is issued.
```

Comment:
Example: cluster A died -- since I cannot demote the image, I need to force promote the image in the other cluster.